src/plugins/intel_cpu/src/nodes/executors/acl/acl_fullyconnected.cpp

```diff
@@ -0,0 +1,350 @@
+        return *weightCache->findOrCreate(string_hash, create);
+    }
+
+    DEBUG_LOG("ACLFullyConnectedExecutor: Weights cache is not available");
+    return create();
+}
+
+static bool checkAndInitPostOps(const PostOps &postOps, arm_compute::FullyConnectedLayerInfo& fullyConnectedLayerInfo) {
+    // Add postops
+    if (!postOps.empty() && postOps.size() == 1) {
+        if (const auto activation = std::dynamic_pointer_cast<ActivationPostOp>(postOps[0])) {
+            return getActivationLayerInfo(convertToEltwiseAlgorithm(activation->type()),
+                                          fullyConnectedLayerInfo.activation_info,
+                                          activation->alpha(),
+                                          activation->beta(),
+                                          activation->gamma());
+        }
+    }
+    return false;
+}
+
+static void initFCAttrs(const FCAttrs &attrs,
+                        ACLTensorAttrs& aclTensorAttrs,
+                        ACLFCAttrs& aclfcAttrs,
+                        const MemoryArgs &memory,
+                        arm_compute::FullyConnectedLayerInfo& fullyConnectedLayerInfo,
+                        const PostOps &postOps) {
+    aclTensorAttrs.hasLayoutTypeNHWC = memory.at(ARG_SRC)->getDescPtr()->hasLayoutType(LayoutType::nspc);
+    fullyConnectedLayerInfo.weights_trained_layout = getAclDataLayoutByMemoryDesc(memory.at(ARG_WEI)->getDescPtr());
+    aclfcAttrs.inputPrecision = memory.at(ARG_SRC)->getDescPtr()->getPrecision();
+    fullyConnectedLayerInfo.transpose_weights = false;
+    aclfcAttrs.weightsNonTransposed = attrs.weightsNonTransposed;
+
+    checkAndInitPostOps(postOps, fullyConnectedLayerInfo);
+
+    if (memory.at(ARG_SRC)->getPrecision() != memory.at(ARG_WEI)->getPrecision()) {
+        aclfcAttrs.isConvertedWeights = true;
+    }
+}
+
+ACLFullyConnectedExecutor::ACLFullyConnectedExecutor(const FCAttrs &attrs,
+                                                     const PostOps &postOps,
+                                                     const MemoryArgs &memory,
+                                                     const ExecutorContext::CPtr context) {
+    initFCAttrs(attrs, aclTensorAttrs, aclfcAttrs, memory, fullyConnectedLayerInfo, postOps);
+    packedWeights = prepareWeightMemory(memory, context, attrs, aclfcAttrs, postOps);
+}
+
+bool ACLFullyConnectedExecutor::supports(const FCConfig &config) {
+    arm_compute::FullyConnectedLayerInfo tmpFullyConnectedLayerInfo;
+    VERIFY(one_of(srcType(config), ov::element::f16, ov::element::f32),     UNSUPPORTED_SRC_PRECISIONS);
+    VERIFY(one_of(weiType(config), ov::element::f16, ov::element::f32),     UNSUPPORTED_WEI_PRECISIONS);
+    VERIFY(postOpsNumbers(config) < 2,                                      UNSUPPORTED_NUMBER_OF_POSTOPS);
+    VERIFY(checkAndInitPostOps(config.postOps, tmpFullyConnectedLayerInfo), UNSUPPORTED_TYPE_OF_POSTOPS);
```

Comment:
Is there a way to provide supported activations before activation fusing? If not, those it mean, that we use reference `FullyConnected` implementation if not supported activation is fused, instead avoid fusing and use ACL `FullyConnected` implementation and execute not supported activation separatelly?

Reply:
@eshoguli corrected